cmake/dts.cmake

```diff
@@ -188,6 +188,13 @@ if(SUPPORTS_DTS)
   if (check)
     set(DTC_NO_WARN_UNIT_ADDR "-Wno-unique_unit_address")
   endif()
+  foreach(extra_opt ${EXTRA_DTC_FLAGS})
+    check_dtc_flag(${extra_opt} check)
+    if (check)
+      list(APPEND VALID_EXTRA_DTC_FLAGS ${extra_opt})
+    endif()
+    endforeach()
```

Comment:
Nits: alignment looks a bit off here and you might want to `set(VALID_EXTRA_DTC_FLAGS)` at the top of the loop just for hygiene.